source/d1min.cpp

```diff
@@ -58,6 +59,42 @@ bool D1Min::load(QString filePath, quint16 subtileCount)
     return true;
 }
 
+bool D1Min::save(SaveAsParam *params)
+{
+    QString filePath = this->getFilePath();
+    if (params != nullptr && !params->minFilePath.isEmpty()) {
+        filePath = params->minFilePath;
+        if (QFile::exists(filePath)) {
+            QMessageBox::StandardButton reply;
+            reply = QMessageBox::question(nullptr, "Confirmation", "Are you sure you want to overwrite the MIN file?", QMessageBox::Yes | QMessageBox::No);
+            if (reply != QMessageBox::Yes) {
+                return false;
+            }
+        }
+    }
+
+    QFile outFile = QFile(filePath);
+    if (!outFile.open(QIODevice::WriteOnly | QFile::Truncate)) {
+        QMessageBox::critical(nullptr, "Error", "Failed open file: " + filePath);
+        return false;
+    }
+
+    // write to file
+    QDataStream out(&outFile);
+    out.setByteOrder(QDataStream::LittleEndian);
+    for (int i = 0; i < this->celFrameIndices.size(); i++) {
+        QList<quint16> &celFrameIndicesList = this->celFrameIndices[i];
+        for (int j = 0; j < celFrameIndicesList.count(); j++) {
+            quint16 writeWord = celFrameIndicesList[j];
+            writeWord |= ((quint16)this->getFrameType(writeWord)) << 12;
+            out << writeWord;
+        }
+    }
+
+    this->minFilePath = filePath; // this->load(filePath, subtileCount);
```

Comment:
I'm a little confused by this comment, is it just to say that we are updating the value originally set in `load()`?

Reply:
at the moment overwriting the minFilePath is enough, but in the future a whole (re-)load might be necessary (similar to d1cel.cpp)